### PR TITLE
Porting a Maui test MauiXDataFormatsTests to xUnit

### PR DESCRIPTION
**Pull request by Lydia-Shi**
Hi @RussKie and @AdamYoblick , could you please take a look at this pull and approve it if the changes are ok?

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/winforms/pull/2415)

### CODE REVIEW

File: src/System.Windows.Forms/tests/IntegrationTests/MauiTests/MauiXDataFormatsTests/MauiXDataFormatsTests.cs [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+            p.log.WriteLine("Format type number : " + iFormat);
+            formatObj = DataFormats.GetFormat(iFormat);
+
+            return new ScenarioResult(formatObj.Id == iFormat, "Unexpected format type : " + formatObj.Id);
+        }
+
+        private string Get_DataFormat_String(TParams p)
+        {
+            string[] strFormats = new string[] { "UnicodeText", "Text", "Bitmap", "MetafilePict", "EnhancedMetafile", "DIF",
+                                     "TIFF", "OEMText", "DIB", "Palette", "PenData", "RIFF", "WaveAudio", "SymbolicLink",
```
Comment (AdamYoblick) [inline src/System.Windows.Forms/tests/IntegrationTests/MauiTests/MauiXDataFormatsTests/MauiXDataFormatsTests.cs]: What's going on with the whitespace here?

File: src/System.Windows.Forms/tests/IntegrationTests/MauiTests/MauiXDataFormatsTests/MauiXDataFormatsTests.cs [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+                                     "FileDrop", "Locale","Test1", "Test2", "Test3" };
+
+            int iRandNum = p.ru.GetRange(0, strFormats.Length - 1);
+            return strFormats[iRandNum];
+        }
+
+        private int Get_DataFormat_Int(TParams p)
+        {
+            int[] iFormatNum = new int[] {  win.CF_UNICODETEXT, win.CF_TEXT, win.CF_BITMAP, win.CF_METAFILEPICT, win.CF_ENHMETAFILE,
```
Comment (AdamYoblick) [inline src/System.Windows.Forms/tests/IntegrationTests/MauiTests/MauiXDataFormatsTests/MauiXDataFormatsTests.cs]: Same as above

File: src/System.Windows.Forms/tests/IntegrationTests/MauiTests/MauiXDataFormatsTests/MauiXDataFormatsTests.cs [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+                         win.CF_DIF, win.CF_TIFF, win.CF_OEMTEXT, win.CF_DIB, win.CF_PALETTE, win.CF_PENDATA, win.CF_RIFF,
+                         win.CF_WAVE, win.CF_SYLK, win.CF_HDROP, win.CF_LOCALE, 100, 200, 300 };
+
+            int iRandNum = p.ru.GetRange(0, iFormatNum.Length - 1);
+            return iFormatNum[iRandNum];
+        }
+
+        private Boolean Verify_Format_Type(string strExp, string strOrg)
+        {
+            if (String.Compare(strExp.Substring(strExp.IndexOf(":") + 1), strOrg, true) == 0)
```
Comment (AdamYoblick) [inline src/System.Windows.Forms/tests/IntegrationTests/MauiTests/MauiXDataFormatsTests/MauiXDataFormatsTests.cs]: Please split this statement up, this is confusing to follow.

**Comment by AdamYoblick** [comment]
Also the build is broken. Did you add any weird characters to the solution or the csproj?

**Comment by Lydia-Shi** [comment]
Hi @AdamYoblick, Thanks for your comment. I have changed the code follow your suggestion and resolved the build error. Could you please take a look at this pull again?

**Comment by Lydia-Shi** [comment]
Hi @RussKie , could you please take a look at this pull and approve it if the changes are ok?

**Comment by RussKie** [comment]
Thank you @Lydia-Shi for the PR, however I am not convinced this port is overall beneficial.

We already have a number of tests covering the area:
image.
Specifically we have tests for `DataFormats` located here: https://github.com/dotnet/winforms/blob/3e772d4a4a7ded0381c3d79d848188fb3b70baa4/src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/DataFormatsTests.cs
I believe these tests provide a good coverage. If you think some scenarios are missing from those tests, please add those scenarios to the unit tests suite.

Thank you

**Comment by Lydia-Shi** [comment]
Thank you @RussKie for your suggestion.

**Pull request closed without merging**